test/Table_Tests/src/Database/Upload_Spec.enso

```diff
@@ -205,6 +217,25 @@ spec make_new_connection prefix persistent_connector=True =
                 r1 = connection.create_table (Name_Generator.random_name "creating-table") structure=[Column_Description.Value "X" Value_Type.Integer] primary_key=[0] temporary=True
                 r1.should_fail_with Illegal_Argument
 
+        Test.specify "should not issue a DELETE statement for the original table name in dry run mode, even if the table does not exist" <|
+            original_table_name = Name_Generator.random_name "no-delete-test"
+
+            log_file = enso_project.data / "transient" / "sql.log"
+            log_file.delete_if_exists
+            Test_Environment.unsafe_with_environment_override "ENSO_SQL_LOG_PATH" log_file.absolute.path <|
+                Context.Output.with_disabled <|
+                    t1 = connection.create_table original_table_name structure=[Column_Description.Value "X" Value_Type.Integer] temporary=True
+                    t1.column_names . should_equal ["X"]
+
+            logs = log_file.read Plain_Text . lines
+            deletes = logs.filter (_.contains "DROP")
+            wrapped_name = connection.dialect.wrap_identifier original_table_name
+            deletes.each line->
+                if line.contains wrapped_name then
+                    Test.fail "The log file contains a dangerous DELETE statement for the original table: "+line
+
+            log_file.delete_if_exists
```

Comment:
Very specific test to check that we do not issue a bad DROP even if the table to drop does not exist.

Mostly to ensure this bug got fixed and avoid regressions, and to demonstrate testing using the  `ENSO_SQL_LOG_PATH` feature.